### ISSUE TITLE
add OTS to noto_lint

### ISSUE THREAD

**Issue by xiangyexiao** [googlei18n/nototools#302]
Should we add opentype sanity check to our lint?

**Comment by xiangyexiao** [comment]
@brawer 

This is something required by Google Fonts.

**Comment by brawer** [comment]
Should the linter just check that sanitization doesn’t fail, or should it enforce that the sanitizer’s result is bit-wise identical to the font? Is the latter even realistic? If it’s unrealistic to enforce bit-wise identity after sanitization, should we run the sanitization as part of the build pipeline instead of the linter, and do all quality checks on the post-sanitization fonts? (Should the sanitizer strip off any data, we’ll need to fix it anyway).
https://github.com/khaledhosny/ots/blob/master/docs/DesignDoc.md

**Comment by dougfelt** [comment]
Seems best to run it as part of the build.

**Comment by behdad** [comment]
OTS output is not preferable to the pipeline.  We just need to make sure it doesn't barf on our output.  If it does, equally often times it's an OTS bug.

**Comment by dougfelt** [comment]
@jungshik, I thought Chrome used OTS when it loaded fonts as a sanity check.  This would seem to mean that we need to ensure our fonts pass it, even if the fault is with OTS, and so maybe we should do this as a build validation step.  Behdad seems skeptical of the utility of this, though.

**Comment by khaledhosny** [comment]
I think what @behdad is saying is that you should make sure the fonts pass OTS check, but don’t use the output fonts from OTS (as @brawer was suggesting) since OTS will sometimes strip things it thinks not needed by web browsers.

**Comment by davelab6** [comment]
I agree with Behdad about not using the lint run as part of the build
process; for non-fontmake-built fonts, I think roundtripping through
fonttools is good to do in the post processing build process, and will
catch any major badness and often compact the files. But the fonts returned
by ots are a bit more likely to be butchered.

So my suggestion is that ots become an integral step of Noto lint; it's
already integral to fontbakery-check-TTF; that is, run ots as a lint
process (checking return is 0) instead of a build process.

**Comment by xiangyexiao** [comment]
It makes more sense to me to have OTS as a step in lint, rather than the font building process. We just need to make sure our fonts pass OTS check.
